File: 05/hash_table_voted.py
voted = {}

# get 함수는 해시 테이블 안에 "tom"이라는 key가 있으면
# 그 key에 해당하는 value를 반환한다.
# 만약 없다면, None을 반환한다.
value = voted.get("tom")

def check_voter(name):
    if voted.get(name):
        print("쫓아내세요!")
    else:
        voted[name] = True
        print("투표하게 하세요.")


check_voter("tom")
check_voter("mike")
check_voter("mike")

File: 05/hashTableVoted.js
const voted = {};

const value = voted.hasOwnProperty("tom");
console.log(value); // "tom"이라는 property가 없으므로 false를 내놓는다.

const check_voter = (name) => {
	if (voted.hasOwnProperty(name)) {
		console.log("쫓아내세요!");
	} else {
		voted[name] = true;
		console.log("투표하게 하세요.");
	}
};

check_voter("tom");
console.log(voted);
check_voter("mike");
console.log(voted);
check_voter("mike"); // 이미 voted 안에 mike가 존재하므로 "쫓아내세요!"가 출력된다.
